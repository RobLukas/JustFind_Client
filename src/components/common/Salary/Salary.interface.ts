interface SalaryProps {
  salary: {
    from: number;
    to: number;
    currency: string;
  };
}

export default SalaryProps;
