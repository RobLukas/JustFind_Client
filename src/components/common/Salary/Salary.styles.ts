import styled from 'styled-components';

const SalaryWrapper = styled.div({
  marginLeft: 'auto',
  display: 'flex',
  alignItems: 'center',
});

const TitleSalary = styled.span({
  '@media (max-width: 1020px)': {
    lineHeight: '11px',
  },
  color: '#51b368',
  verticalAlign: 'middle',
  textAlign: 'right',
  fontSize: 16,
  whiteSpace: 'nowrap',
});

export default {
  SalaryWrapper,
  TitleSalary,
};
