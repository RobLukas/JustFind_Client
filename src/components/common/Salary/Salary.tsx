import React, { FC } from 'react';
import SalaryProps from './Salary.interface';
import S from './Salary.styles';

const Salary: FC<SalaryProps> = ({ salary }: SalaryProps) => {
  const { from, to, currency } = salary;

  return <S.TitleSalary>{`${from} - ${to} ${currency}`}</S.TitleSalary>;
};

export default Salary;
