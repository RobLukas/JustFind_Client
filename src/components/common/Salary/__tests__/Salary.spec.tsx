import React from 'react';
import { cleanup, render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Salary from '../Salary';
import SalaryProps from '../Salary.interface';

describe('<Salary />', () => {
  afterEach(cleanup);
  it('renders correctly', () => {
    const props: SalaryProps = {
      salary: {
        from: 10000,
        to: 20000,
        currency: 'PLN',
      },
    };
    const tree = renderer.create(<Salary {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders with salary props', () => {
    const props: SalaryProps = {
      salary: {
        from: 1000,
        to: 2000,
        currency: 'PLN',
      },
    };
    const { container, getByText } = render(<Salary {...props} />);
    expect(getByText('1000 - 2000 PLN')).toBeInTheDocument();
    expect(container.firstChild).toMatchInlineSnapshot(
      `
      <span
        class="sc-gtssRu ibqKak"
      >
        1000 - 2000 PLN
      </span>
    `,
    );
  });
});
