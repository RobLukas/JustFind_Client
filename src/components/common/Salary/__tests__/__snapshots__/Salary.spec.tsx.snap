// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Salary /> renders correctly 1`] = `
<span
  className="sc-gtssRu ibqKak"
>
  10000 - 20000 PLN
</span>
`;
